pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/AvroSchemaKafkaAvroMessageDecoder.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.compat.tests;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Set;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.pinot.plugin.inputformat.avro.AvroRecordExtractor;
+import org.apache.pinot.spi.data.readers.GenericRow;
+import org.apache.pinot.spi.data.readers.RecordExtractor;
+import org.apache.pinot.spi.stream.StreamMessageDecoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class AvroSchemaKafkaAvroMessageDecoder implements StreamMessageDecoder<byte[]> {
```

Comment:
Can we pull this from ClusterTest?

Reply:
`ClusterTest` uses `AvroSchemaKafkaAvroMessageDecoder` which is kind of hacky. Currently when we run the integration tests, server needs to know the avro schema when creating the test table. `AvroSchemaKafkaAvroMessageDecoder` uses a static field to pass that information from test main thread to the server thread. For this case, server is run as a process, the same approach won't work.

Reply:
Hmm, maybe we can update current `AvroSchemaKafkaAvroMessageDecoder` logic to use the same approach here (pull schema info from `props`), and use the updated `AvroSchemaKafkaAvroMessageDecoder` for both integration & compatiblity tests.

Reply:
The servers need to be provided with the right Kafka classes to read the kafka incoming messages. Can you look into modifying the classpath for starting servers to include the appropriate classes? Or, is it possible to use the default avro decoder that comes with kafka . See how quickstart works

Reply:
The default avro decoder needs an avro schema service so server can fetch the schema for decoding.
For quickstart, we add pinot-plugin to class path to run pinot server, and JSON decoder is used.
For our case, seems like the `compCheck.sh` is adding integration-tests to classpath_prefix. Not sure it's this classpath_prefix is picked up correctly, I can double check if it works later today.